clouddriver-eureka/src/main/groovy/com/netflix/spinnaker/clouddriver/eureka/deploy/ops/AbstractEurekaSupport.groovy

```diff
@@ -216,23 +233,35 @@ abstract class AbstractEurekaSupport {
 
         retryCount++
         sleep(getDiscoveryRetryMs());
-      } catch (RetrofitError re) {
+      } catch (NotFoundException e) {
         if (retryCount >= (maxRetries - 1)) {
-          throw re
+          throw e
         }
-
-        AbstractEurekaSupport.log.debug("[$phaseName] - Failed calling external service ${re.message}")
-
-        if (re.kind == RetrofitError.Kind.NETWORK || re.response.status == 404 || re.response.status == 406) {
-          retryCount++
+        retryCount++
+        AbstractEurekaSupport.log.debug("[$phaseName] - Failed calling external service ${e.getMessage()}")
+        sleep(getDiscoveryRetryMs())
+      } catch (SpinnakerHttpException e) {
+        if (retryCount >= (maxRetries - 1)) {
+          throw e
+        }
+        retryCount++
+        AbstractEurekaSupport.log.debug("[$phaseName] - Failed calling external service ${e.getMessage()}")
+        if (e.getResponse().getStatus() == HttpStatus.NOT_ACCEPTABLE.value()) {
           sleep(getDiscoveryRetryMs())
-        } else if (re.response.status >= 500) {
+        } else if (e.getResponse().getStatus() >= 500) {
           // automatically retry on server errors (but wait a little longer between attempts)
           sleep(getDiscoveryRetryMs() * 10)
-          retryCount++
-        } else {
-          throw re
         }
+      } catch (SpinnakerNetworkException e) {
+        if (retryCount >= (maxRetries - 1)) {
+          throw e
+        }
+        retryCount++
+        AbstractEurekaSupport.log.debug("[$phaseName] - Failed calling external service ${e.getMessage()}")
+        sleep(getDiscoveryRetryMs())
+      } catch (SpinnakerServerException e) {
+        AbstractEurekaSupport.log.debug("[$phaseName] - Failed calling external service ${e.getMessage()}")
+        throw e
```

Comment:
Not a huge deal, logging and rethrowing is generally an anti-pattern... either handle the exception or let the caller decide what to do with it. Most likely the caller is going to log this somewhere anyway...